File: src/main/java/main/ui/renderer/HelpTableCellRenderer.java
package main.ui.renderer;

import org.jetbrains.annotations.NotNull;

import javax.swing.*;
import javax.swing.table.DefaultTableCellRenderer;
import java.awt.*;

/**
 *
 * @author Bunnyspa
 */
public class HelpTableCellRenderer extends DefaultTableCellRenderer {

    private final int rowHeaderNum, colHeaderNum;

    public HelpTableCellRenderer(int rowHeaderNum, int colHeaderNum) {
        this.rowHeaderNum = rowHeaderNum;
        this.colHeaderNum = colHeaderNum;
    }

    @NotNull
    @Override
    public Component getTableCellRendererComponent(JTable table, Object value, boolean isSelected, boolean hasFocus, int row, int column) {
        DefaultTableCellRenderer cr = (DefaultTableCellRenderer) super.getTableCellRendererComponent(table, value, isSelected, hasFocus, row, column);
        cr.setHorizontalAlignment(CENTER);

        if (row < colHeaderNum || (column < rowHeaderNum)) {
            cr.setBackground(SystemColor.control);
        } else {
            cr.setBackground(Color.WHITE);
        }

        return cr;
    }
}

File: src/main/java/main/ui/renderer/HelpTableCellRenderer.kt
package main.ui.renderer

import java.awt.Color
import java.awt.Component
import java.awt.SystemColor
import javax.swing.JTable
import javax.swing.table.DefaultTableCellRenderer

/**
 *
 * @author Bunnyspa
 */
class HelpTableCellRenderer(private val rowHeaderNum: Int, private val colHeaderNum: Int) : DefaultTableCellRenderer() {
    override fun getTableCellRendererComponent(
        table: JTable,
        value: Any,
        isSelected: Boolean,
        hasFocus: Boolean,
        row: Int,
        column: Int
    ): Component {
        val cr = super.getTableCellRendererComponent(
            table,
            value,
            isSelected,
            hasFocus,
            row,
            column
        ) as DefaultTableCellRenderer
        cr.horizontalAlignment = CENTER
        if (row < colHeaderNum || column < rowHeaderNum) {
            cr.background = SystemColor.control
        } else {
            cr.background = Color.WHITE
        }
        return cr
    }
}